test/e2e/storage/testsuites/provisioning.go

```diff
@@ -289,11 +292,8 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
 				myTestConfig := testConfig
 				myTestConfig.Prefix = fmt.Sprintf("%s-%d", myTestConfig.Prefix, i)
 
-				// Each go routine must have its own testCase copy to store their claim
-				myTestCase := *l.testCase
-				myTestCase.Claim = myTestCase.Claim.DeepCopy()
-				myTestCase.Class = nil // Do not create/delete the storage class in TestDynamicProvisioning, it already exists.
-				myTestCase.PvCheck = func(claim *v1.PersistentVolumeClaim) {
+				t := *l.testCase
+				t.PvCheck = func(claim *v1.PersistentVolumeClaim) {
```

Comment:
why remove Claim.DeepCopy()?

Reply:
Also I don't know why class was set to nil before. The author @jsafrane https://github.com/kubernetes/kubernetes/commit/1f9f2390cb57832f5fbd02e53d39023b81500cc2 specifically made a comment about storageclass already exist? Is it trying to test default storage class behavior?

Reply:
t.Claim is just a spec without a name, `TestDynamicProvisioning` will create the actual object, from what I understand this will happen as long as the object created is unique and it is because it doesn't have a name

for `t.Class = nil` it might have been a mistake because the test wasn't using the class setup before the goroutines (it was instead using the default storageclass if it existed), @msau42 clarified that inside testsuites we don't have tests for the default storageclass

Reply:
I see. The claim is using a GenerateName which provides the prefix and when creating PVC, it will generate a unique dynamically.

Reply:
Ack to the storage class changes in this PR - it's better to create a new one for each test inside the testsuites.

Reply:
Regarding `t.Claim`, does every caller use GenerateName? If we want to make that assumption, then it may be better to assert on it so we catch any future changes.

Reply:
yes, all of the callers use `e2epv.MakePersistentVolumeClaim` with an empty `Name` but with a `NamePrefix`, I added an assertion in `TestDynamicProvisioning` to ensure that `NamePrefix` is set (although it could be overriden by setting Name but no test does this at this moment)